### PR TITLE
feat: first attempt at firecracker bootstrapping

### PR DESCRIPTION
**Pull request by johnrwatson**
Added basic configuration of ubuntu metal node in EC2 for use with Firecracker.

Use the following userdata to fully configure the node. Content can be found in the `//firecracker-data/` directory with 5000 taps pre-provisioned.

```
#!/bin/bash

set -eo pipefail

# Add an example variables file
cat << HEREDOC > /tmp/variables.txt
CONFIGURATION_MANAGEMENT_TOOL="shell"
AUTOMATED="true"
HEREDOC

curl -s https://raw.githubusercontent.com/systeminit/si-firecracker-config/main/orchestrate-install.sh | bash
```

### CODE REVIEW

File: machine-userdata.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+# Add an example variables file
+cat << HEREDOC > /tmp/variables.txt
+CONFIGURATION_MANAGEMENT_TOOL="shell"
+AUTOMATED="true"
+HEREDOC
+
+curl https://raw.githubusercontent.com/systeminit/si-firecracker-config/main/orchestrate-install.sh > /tmp/orchestrate-install.sh
+chmod a+x /tmp/orchestrate-install.sh
+/bin/bash /tmp/orchestrate-install.sh /tmp/variables.txt
```
Comment (sprutton1) [inline machine-userdata.sh:13]: Super nit: If you fell back to a default filename you could just invoke this via 
```bash
curl -s https://raw.githubusercontent.com/systeminit/si-firecracker-config/main/orchestrate-install.sh | bash
```

Comment (johnrwatson) [reply machine-userdata.sh]: Super nit accepted!

File: orchestrate-install.sh [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            echo "Error: Something went wrong, OS_VARIANT determined to be: $OS_VARIANT (unsupported)" && exit 1
+        ;;
+    esac
+
+    [[ $? != 0 ]] && echo "Error: Exit code $? returned during installation; see above error log for information"
+
+    return 0
+
+}
+execute_configuration_management() {
```
Comment (sprutton1) [inline orchestrate-install.sh:87]: Nit: this fine for now, but we should consider decomposing this around the action types, i.e. `setup_users()`, `get_binaries()` to make it a bit more readable

**Review by sprutton1 — APPROVED** [review]

File: orchestrate-install.sh (hunk 2 above)
Comment (johnrwatson) [reply orchestrate-install.sh:87]: Nice, yep fair shout. I'm not even 100% sure if we will need the cleanup or prereq function based on how simple the setup is

**Pull request merged**